### PR TITLE
Add dbcppp/3.2.6

### PR DESCRIPTION
**Pull request by Nomalah**
Specify library name and version:  **dbcppp/3.2.6**

Requesting to add: [dbcppp](https://github.com/xR3b0rn/dbcppp)

dbcppp is very useful when using .dbc files to decode CAN messages

It did take me some time, but the CMake from the project should be patched for conan. I did have to drop some support from the project due to it's configuration. I also was not able to get dbcppp:shared=True working on windows (untested on Linux), so I have disabled it.

---

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/dbcppp/all/conandata.yml [hunk 1]
```diff
@@ -0,0 +1,9 @@
+sources:
+  "3.2.6":
+    url: https://github.com/xR3b0rn/dbcppp/archive/refs/tags/v3.2.6.tar.gz
+    sha256: "e54829abf503abbc95513629e9318450224c1b22c0fcee4febc16cbfd73afafb"
+patches:
+  "3.2.6":
+    - patch_file: "patches/0001-dep-cmake-rem-kcd.patch"
+      patch_type: conan
+      patch_description: Disable use of KCD to ensure compatibility with conan
```
Comment (uilianries) [inline recipes/dbcppp/all/conandata.yml:9]:
```suggestion
      patch_type: "conan"
      patch_description: "Disable use of KCD to ensure compatibility with conan"
```

please, keep everything as string. We had parser errors in past, it's rare but happens.

File: recipes/dbcppp/all/conanfile.py [hunk 2]
```diff
@@ -0,0 +1,113 @@
+from conan import ConanFile
+from conan.errors import ConanInvalidConfiguration
+from conan.tools.microsoft import check_min_vs, is_msvc_static_runtime, is_msvc
+from conan.tools.files import apply_conandata_patches, export_conandata_patches, get, copy, rm, rmdir
+from conan.tools.build import check_min_cppstd
+from conan.tools.scm import Version, Git
```
Comment (uilianries) [inline recipes/dbcppp/all/conanfile.py:6]:
```suggestion
from conan.tools.scm import Version
```
Not used

File: recipes/dbcppp/all/conanfile.py [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+                del self.options.fPIC
+            except Exception:
+                pass
+
+    def layout(self):
+        cmake_layout(self)
+
+    def requirements(self):
+        if self.options.with_tools:
+            self.requires("cxxopts/2.2.1")
```
Comment (uilianries) [inline recipes/dbcppp/all/conanfile.py:62]: Is not possible to use 3.0.0?

File: recipes/dbcppp/all/conanfile.py [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+        if not is_msvc(self):
+            minimum_version = self._compilers_minimum_version.get(str(self.info.settings.compiler), False)
+            if minimum_version and Version(self.info.settings.compiler.version) < minimum_version:
+                raise ConanInvalidConfiguration(
+                    f"{self.ref} requires C++{self._minimum_cpp_standard}, which your compiler does not support."
+                )
+
+    def source(self):
+        get(self, **self.conan_data["sources"][str(self.version)],destination=self.source_folder, strip_root=True)
+        rm(self, "KCD2Network.cpp", self.source_folder, recursive=True) # Cannot support KCD because of weird dll issues
```
Comment (uilianries) [inline recipes/dbcppp/all/conanfile.py:80]:
```suggestion
```
Please move it to patch, or to be executed after apply_conandata_patches. source() should preserve all the original folder, no changes are allowed here.

File: recipes/dbcppp/all/conanfile.py [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+        cmake = CMake(self)
+        cmake.configure()
+        cmake.build()
+
+    def package(self):
+        copy(self, pattern="LICENSE", dst=os.path.join(self.package_folder, "licenses"), src=self.source_folder)
+        cmake = CMake(self)
+        cmake.install()
+
+        rmdir(self, os.path.join(self.package_folder, "lib", "pkgconfig"))
```
Comment (uilianries) [inline recipes/dbcppp/all/conanfile.py:104]: PLease, remove those methods which are not needed. It only provides cmake files: https://github.com/xR3b0rn/dbcppp/blob/v3.2.6/CMakeLists.txt#L48

File: recipes/dbcppp/all/patches/0001-dep-cmake-rem-kcd.patch [hunk 6]
```diff
@@ -0,0 +1,316 @@
+diff --git a/CMakeLists.txt b/CMakeLists.txt
+index d905b29..ea94456 100644
+--- a/CMakeLists.txt
++++ b/CMakeLists.txt
+@@ -5,53 +5,17 @@ project("libdbcppp" VERSION 0.1.0)
+ 
+ option(build_tests "Build tests" ON)
+ option(build_examples "Build examples" ON)
++option(build_tools "Build examples" ON)
```
Comment (uilianries) [inline recipes/dbcppp/all/patches/0001-dep-cmake-rem-kcd.patch:9]: Could you please open a PR to the upstream providing this PR?

```suggestion
+option(build_tools "Build tools" ON)
```

**Review by uilianries — COMMENTED** [review]
Thank you for your contribution!

File: recipes/dbcppp/all/conanfile.py (hunk 5 above)
Comment (Nomalah) [reply recipes/dbcppp/all/conanfile.py:104]: It is used from here: https://github.com/xR3b0rn/dbcppp/blob/54ce78ad53442db94e03d16a4fd5f3b339dab193/src/libdbcppp/CMakeLists.txt#L51
changes to make it properly package with conan are in the patch.

File: recipes/dbcppp/all/conanfile.py (hunk 3 above)
Comment (Nomalah) [reply recipes/dbcppp/all/conanfile.py]: Tested with 3.0.0 and works.

File: recipes/dbcppp/all/patches/0001-dep-cmake-rem-kcd.patch (hunk 6 above)
Comment (Nomalah) [reply recipes/dbcppp/all/patches/0001-dep-cmake-rem-kcd.patch:9]: Don't quite understand. do you mean use the patch for a PR in upstream?

Comment (prince-chrismc) [reply recipes/dbcppp/all/patches/0001-dep-cmake-rem-kcd.patch:9]: Yes! When ever we see contributions that would benefit everyone we like to encourage sharing them

File: recipes/dbcppp/all/conanfile.py [hunk 7]
```diff
@@ -0,0 +1,113 @@
...
+
+    def generate(self):
+        tc = CMakeToolchain(self)
+        if is_msvc(self):
+            tc.variables["USE_MSVC_RUNTIME_LIBRARY_DLL"] = not is_msvc_static_runtime(self)
+        tc.variables["build_tests"] = False
+        tc.variables["build_examples"] = False
+        tc.variables["build_tools"] = self.options.with_tools
+        tc.generate()
+        deps = CMakeDeps(self)
```
Comment (prince-chrismc) [inline recipes/dbcppp/all/conanfile.py:89]: If tools here? Otherwise it's not needed?

File: recipes/dbcppp/all/conanfile.py [hunk 8]
```diff
@@ -0,0 +1,113 @@
...
+        cmake.configure()
+        cmake.build()
+
+    def package(self):
+        copy(self, pattern="LICENSE", dst=os.path.join(self.package_folder, "licenses"), src=self.source_folder)
+        cmake = CMake(self)
+        cmake.install()
+
+        rmdir(self, os.path.join(self.package_folder, "lib", "pkgconfig"))
+        rmdir(self, os.path.join(self.package_folder, "lib", "cmake"))
```
Comment (prince-chrismc) [inline recipes/dbcppp/all/conanfile.py:105]: Is this from the template pr the hooks?

It it's needed for the hooks there's a good clue we are missing some properties in the package info https://github.com/conan-io/conan-center-index/blob/master/docs/v2_migration.md#new-cpp_info-set_property-model

File: recipes/dbcppp/all/patches/0001-dep-cmake-rem-kcd.patch [hunk 9]
```diff
@@ -0,0 +1,316 @@
...
++set(CMAKE_WINDOWS_EXPORT_ALL_SYMBOLS ON)
+ 
+ if (CMAKE_CXX_COMPILER_ID MATCHES "MSVC")
+ 	add_definitions("/bigobj")
+ endif()
+ 
+ include_directories("include")
+-include_directories("third-party/libxml2/include")
+-include_directories("third-party/libxmlmm/libxmlmm")
+ include_directories("third-party/boost")
```
Comment (prince-chrismc) [inline recipes/dbcppp/all/patches/0001-dep-cmake-rem-kcd.patch:24]: It's not using boost from conan?

Comment (Nomalah) [reply recipes/dbcppp/all/patches/0001-dep-cmake-rem-kcd.patch:24]: None of the public header files use boost, so right now I am using the boost that's included in their repo. Should I change this to one provided by conan? It's a pretty heavy dependency.

Comment (prince-chrismc) [reply recipes/dbcppp/all/patches/0001-dep-cmake-rem-kcd.patch:24]: The idea is that all the dependencies are managed by Conan - this allows the client to report conflicts and avoid linker problems mixing versions

Comment (Nomalah) [reply recipes/dbcppp/all/patches/0001-dep-cmake-rem-kcd.patch]: Fixed!

**Review by uilianries — APPROVED** [review]

**Review by prince-chrismc — APPROVED** [review]

**Pull request merged**